liferay-echo/liferay-portal
# LPS-151738 Don't do anything if the FF is not enabled

In this PR we are adding the ability to add `data-lfr-styles` to the fragment to inject there the common styles class. 

### Context
In https://issues.liferay.com/browse/LPS-147783 We moved the common styles from using inline styles to use a class and move those styles to the head of the page. In this story what we are trying to achieve is to apply that class to a specific element inside the fragment. Since we are aplying the styles to an outer div of the fragment, styles like background color or padding can result in weird behaviours, for example in the button fragment, a user would expect that changing the background would apply to the button itself, not the outer container. 

### How are we doing it?

A new processor has been created that inject the class `lfr-layout-structure-item-<item-id>` to the element that contains the `data-lfr-styles` attribute. This processor also includes information in the editableValues to register that this fragment have inner common styles. That information will be used in the page editor and in the view mode to not include the class in the outer div.

### How to test this?

> Enable the Feature flag first: `feature.flag.LPS-132571=true` 

The easier one is to clone the OOTB button fragment and add the `data-lfr-styles`

- Go to Design > Fragments
- Create a new fragment Set
- Go to Basic components set
- In the button fragment, click on the kebab menu and choose "Copy to" and select the new fragment set
- Edit the button and include `data-lfr-styles` in the <a> element:
```html
<div class="component-button text-break">
	<a class="btn btn-${configuration.buttonSize} btn-${configuration.buttonType}" data-lfr-editable-id="link" data-lfr-editable-type="link" href="" id="fragment-${fragmentEntryLinkNamespace}-link" data-lfr-styles>
		Go Somewhere
	</a>
</div>
```
- Create a new content page and add this fragment
- Apply some style
- Try the same with an OOTB button to see the diferences
